tests/integration-tests/conftest.py

```diff
@@ -1193,17 +1193,119 @@ def placement_group_stack(cfn_stacks_factory, request, region):
     cfn_stacks_factory.delete_stack(stack.name, region)
 
 
-@pytest.fixture(scope="class")
-def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_stack: CfnVpcStack):
-    logging.info("Setting up the ODCR stack")
-    odcr_template = Template()
-    odcr_template.set_version()
-    odcr_template.set_description("ODCR stack to test open, targeted, and PG ODCRs")
+def _odcr_azs(vpc_stack):
     public_subnet = vpc_stack.get_public_subnet()
     public_subnets = vpc_stack.get_all_public_subnets()
     default_public_az = boto3.resource("ec2").Subnet(public_subnet).availability_zone
     availability_zone_1 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
     availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[1]).availability_zone
+
+    return {"default_public_az": default_public_az, "az1": availability_zone_1, "az2": availability_zone_2}
+
+
+@pytest.fixture(scope="class")
+def scaling_odcr_stack(
+    request,
+    region,
+    cfn_stacks_factory,
+    vpc_stack: CfnVpcStack,
+):
+    odcr_stack: Union[CfnStack, None] = None
+
+    def _scaling_odcr_stack(instances_count: int):
+        nonlocal odcr_stack
+
+        logging.info("Setting up the ODCR stack for scaling with special cases")
+        odcr_template = Template()
+        odcr_template.set_version()
+        odcr_template.set_description("ODCR stack to test scaling with special cases")
+        default_public_az, availability_zone_1, availability_zone_2 = _odcr_azs(vpc_stack).values()
+
+        first_odcr_instances_count = int(instances_count // 2)
+        scaling_odcr_a = ec2.CapacityReservation(
+            "integTestsScalingOdcrA",
+            AvailabilityZone=default_public_az,
+            InstanceCount=first_odcr_instances_count,
+            InstancePlatform="Linux/UNIX",
+            InstanceType="c5.large",
+            InstanceMatchCriteria="targeted",
+        )
+        scaling_odcr_b = ec2.CapacityReservation(
+            "integTestsScalingOdcrB",
+            AvailabilityZone=default_public_az,
+            InstanceCount=instances_count - first_odcr_instances_count,
+            InstancePlatform="Linux/UNIX",
+            InstanceType="c5.large",
+            InstanceMatchCriteria="targeted",
+        )
+
+        scaling_odcr_group = resourcegroups.Group(
```

Comment:
A single ODCR has a limit of 1000 instances so to support a test requiring more than that we need 2 ODCRs. The splitting was done mainly to distribute the needed capacity across the ODCRs.

An alternative could be to divide the ODCRs into "bins" of 1000 with the last ODCR potentially having <1000. I mostly went for splitting since we're not testing >2000 instances for now. It could be an improvement for later if needed.